plugins/engines/ospray/ispc/render/SimulationRenderer.ispc

```diff
@@ -106,308 +106,253 @@ inline void composite(const varying vec4f& src, varying vec4f& dst,
-
-            //                    Ray lightRay = ray;
-            //                    if (self->softShadows > 0.f)
-            //                        lightRay.dir = normalize(
-            //                            lightSample.dir +
-            //                            self->softShadows *
-            //                                getRandomVector(sample,
-            //                                lightSample.dir,
-            //                                                self->randomNumber));
-            //                    else
-            //                        lightRay.dir = lightSample.dir;
-
-            //                    lightRay.t =
-            //                    length(make_vec3f(self->volumeDimensions));
-            //                    lightRay.t0 = dg.epsilon;
-            //                    lightRay.org = point;
-            //                    lightRay.dir = lightSample.dir;
-
-            //                    shadowIntensity +=
-            //                        getVolumeShadowContribution(self,
-            //                        lightRay, sample);
-            //                }
-            //                const float giAttenuation =
-            //                    1.f - shadowIntensity * self->shadows;
-            //                voxelColor.x = voxelColor.x * giAttenuation;
-            //                voxelColor.y = voxelColor.y * giAttenuation;
-            //                voxelColor.z = voxelColor.z * giAttenuation;
-            //            }
-
-            // Compose final voxel color
-            vec4f voxelColor =
-                clamp(sampleOpacity / volume->samplingRate) *
-                make_vec4f(sampleColor.x, sampleColor.y, sampleColor.z, 1.0f);
-            composite(voxelColor, pathColor, (float)self->volumeSamplesPerRay);
+            volumeSampleColor = volume->transferFunction->getColorForValue(
+                volume->transferFunction, volumeSample);
+
+            // Voxel shading
+            bool shadingEnabled = volume->gradientShadingEnabled;
+            if (volume->singleShade)
+                shadingEnabled &= (samplingCounter == 0);
+            if (shadingEnabled)
+            {
+                vec3f gradient =
+                    safe_normalize(volume->computeGradient(volume, point));
+                // Setup differential geometry for the volume sample point.
+                DifferentialGeometry dg;
+                dg.P = point;
+                dg.epsilon = epsilon;
+                dg.Ns = gradient;
+
+                gradient =
+                    safe_normalize(volume->computeGradient(volume, point));
+
+                // Shading according to computed normal
+                const vec2f s = make_vec2f(0.5f);
+                const Light_SampleRes light =
+                    self->abstract.lights[0]->sample(self->abstract.lights[0],
+                                                     dg, s);
```

Comment:
Maybe just use make_vec2f(0.5f) and remove variable s.

Reply:
I cannot, the sample function needs a reference, so there needs to be a variable there :(